stream.go

```diff
@@ -123,6 +123,9 @@ type ClientStream interface {
 	// calling RecvMsg on the same stream at the same time, but it is not safe
 	// to call SendMsg on the same stream in different goroutines. It is also
 	// not safe to call CloseSend concurrently with SendMsg.
+	//
+	// It is not safe to modify the message after calling SendMsg. Tracing
+	// libraries and stats handlers may use the message lazily.
```

Comment:
Looks good, thank you. One suggestion: as the unsafe usage cases seem to grow, what about considering to list them as bullet points as done above for `SendMsg blocks until`? Eg

>  It is safe to have a goroutine ... at the same time, but it is not safe to:
> - call SendMsg on the same ...
> - call CloseSend concurrently ...
> - modify the message after ...
>